lazyflow/operators/ioOperators/opStreamingHdf5SequenceReaderM.py

```diff
@@ -0,0 +1,240 @@
+                if new_axis not in slice_axes:
+                    break
+
+            if new_axis == '0':
+                # All axes used already.
+                # Stack across first existing axis
+                new_axis = slice_axes[0]
+
+        self._opStacker.Images.resize(0)
+        self._opStacker.Images.resize(num_files)
+        self._opStacker.AxisFlag.setValue(new_axis)
+
+        for opReader in self._readers:
+            opReader.cleanUp()
+
+        self._hdf5Files = []
+        self._readers = []
+        for external_path, internal_path, stacker_slot in zip(
+                external_paths, internal_paths, self._opStacker.Images):
+            opReader = OpStreamingHdf5Reader(parent=self)
+            try:
+                hdfFile = h5py.File(external_path, 'r')
+                opReader.InternalPath.setValue(internal_path)
+                opReader.Hdf5File.setValue(hdfFile)
+            except RuntimeError as e:
+                logger.error(str(e))
+                raise OpStreamingHdf5SequenceReaderM.FileOpenError(external_path)
+            else:
+                stacker_slot.connect(opReader.OutputImage)
+                self._readers.append(opReader)
+                self._hdf5Files.append(hdfFile)
+
+    def propagateDirty(self, slot, subindex, roi):
+        if slot == self.SequenceAxis or slot == self.GlobString:
+            self.OutputImage.setDirty(slice(None))
+
+    @staticmethod
+    def expandGlobStrings(globStrings):
+        """Matches a list of globStrings to internal paths of files
+
+        Args:
+            globStrings: string. glob or path strings delimited by os.pathsep
+
+        Returns:
+            List of internal paths matching the globstrings that were found in
+            the provided h5py.File object
+
+        """
+        external_paths = []
+        internal_paths = []
+        # Parse list into separate globstrings and combine them
+        for globString in globStrings.split(os.path.pathsep):
+            s = globString.strip()
+            components = PathComponents(s)
+            tmp = sorted(glob.glob(components.externalPath))
+            external_paths.extend(tmp)
+            internal_paths.extend(
+                [components.internalPath for i in xrange(len(tmp))])
+        return external_paths, internal_paths
+
+    @staticmethod
+    def checkGlobString(globString):
+        """Checks whether globString is valid for this class
+
+        Rules for globString:
+
+        * must contain multiple external paths or
+        * placeholder '*' in external path
+        * must not contain placeholders in internal paths (for now)
+
+        Args:
+            globString (string): String, one or multiple paths separated with
+              os.path.pathsep and possibly containing '*' as a placeholder.
+
+        Returns:
+            bool: True if rules are met -> this is the right reader
+                  False if rules are not met -> this is not the right reader
```

Comment:
Looks like this docstring is incorrect, since this function never returns `False`, AFAICT.

On a related note, this function (and the analogous one in `OpStreamingHdf5SequenceReaderS`) seems a little weird to me.  It mixes modes of "returning" it's result.  It either **returns** `True` or **raises** an exception.  I think it would be clearer if it behaved in one of the following ways:

1. Return nothing, but raise an exception if there's a problem with the `globString`.  (Returning `True` gives the reader the incorrect impression that this function that the return value is meaningful.)

2. Never raise an exception, but instead simply return `True` or `False` depending on whether or not there's a problem.  Of course, the problem with this approach is that the caller would be given no hint as to *why* the globString is bad.

3. Never raise an exception, but instead return some enum-like code to distinguish between `VALID`, `INVALID_WRONG_FILE_TYPE`, `INVALID_NO_EXTERNAL_PLACEHOLDER`, `INVALID_SAME_FILE`.

I suppose option 1 is the simplest fix.